### ISSUE TITLE
Issue in PYTHON_LIBRARY_HINT

### ISSUE THREAD

**Issue by nim65s** [jrl-umi3218/jrl-cmakemodules#310]
Hi,

https://github.com/jrl-umi3218/jrl-cmakemodules/blob/54ece258eed16da94c7f10979588fcb47b744eb8/python.cmake#L68-L75

I don't understand what is the purpose of this `PYTHON_LIBRARY_HINT` variable, but it seems that it is not used anywhere, and we even have a `TODO: Should be removed later` here:

https://github.com/jrl-umi3218/jrl-cmakemodules/blob/54ece258eed16da94c7f10979588fcb47b744eb8/python.cmake#L98

Moreover, it looks like in cases where the python command fails, the `ERROR_QUIET` implies that the `PYTHON_LIBRARY_HINT` doesn't exist, and therefore the `string(REGEX REPLACE …)` is wrong because it doesn't have the right number of defined arguments:

```
CMake Error at cmake/python.cmake:75 (string):
  string sub-command REGEX, mode REPLACE needs at least 6 arguments total to
  command.
Call Stack (most recent call first):
  CMakeLists.txt:29 (FINDPYTHON)


CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PythonLibs (missing: PYTHON_INCLUDE_DIRS) (found version
  "3.6.9")
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.10/Modules/FindPythonLibs.cmake:262 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/python.cmake:97 (FIND_PACKAGE)
  CMakeLists.txt:29 (FINDPYTHON)
```

This error appears randomly, and we (@florent-lamiraux & I) don't know how to reliably reproduce the error in this `EXECUTE_PROCESS` that seems to cause the issue, because once again, this `ERROR_QUIET` doesn't help to debug.

Suggestion: drop that code.
Ref #262 & #263, where this code was updated, but without clear description or context.

**Comment by jcarpent** [comment]
Sorry for this noise. But the intent of this code was to retrieve the Python lib directly from the python executable (which knows this information). This would be very helpful when someone only provides PYTHON_EXECUTABLE to CMake.
By using this technic, I was able to retrieve all the information in a coherent manner, but after some tests, I realized that, for some Python distributions, this was not working for all the OS.

In conclusion, I think you can drop this code.

**Comment by nim65s** [comment]
I see, thanks.

I think that the only times where providing the executable was not enough for CMake to find the corresponding libs for me were when I did not remove my configuration cache, and therefore an old library was there. 

Then I understand why you wanted to add this feature, but if the current status doesn't help and can even break configuration sometimes, the easiest solution is to delete it.

After that, if the issue occurs again, we can give another try at this feature.
